nssp_demo/forecast_state.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import logging
+import os
+import subprocess
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import numpyro
+import polars as pl
+from prep_data import process_and_save_state
+
+numpyro.set_host_device_count(4)
+
+from fit_model import fit_and_save_model  # noqa
+from generate_predictive import generate_and_save_predictions  # noqa
+
+
+def forecast_denominator(model_dir: Path, n_forecast_days: int) -> None:
+    subprocess.run(
+        [
+            "Rscript",
+            "nssp_demo/forecast_non_target_visits.R",
+            "--model-dir",
+            f"{model_dir}",
+            "--n-forecast-days",
+            f"{n_forecast_days}",
+        ]
+    )
+    return None
+
+
+def postprocess_forecast(model_dir: Path) -> None:
+    subprocess.run(
+        [
+            "Rscript",
+            "nssp_demo/postprocess_state_forecast.R",
+            "--model-dir",
+            f"{model_dir}",
+        ]
+    )
+    return None
+
+
+def main(
+    disease,
+    report_date,
+    state,
+    nssp_data_dir,
+    param_data_dir,
+    output_data_dir,
+    n_training_days,
+    n_forecast_days,
+    n_chains,
+    n_warmup,
+    n_samples,
+):
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+
+    if report_date == "latest":
+        report_date = max(
+            f.stem for f in Path(nssp_data_dir).glob("*.parquet")
+        )
+
+    report_date = datetime.strptime(report_date, "%Y-%m-%d").date()
+
+    logger.info(f"Report date: {report_date}")
+
+    last_training_date = report_date - timedelta(days=1)
+    # + 1 because max date in dataset is report_date - 1
+    first_training_date = last_training_date - timedelta(
+        days=n_training_days - 1
+    )
+
+    datafile = f"{report_date}.parquet"
+    nssp_data = pl.scan_parquet(Path(nssp_data_dir, datafile))
+    param_estimates = pl.scan_parquet(Path(param_data_dir, "prod.parquet"))
+    model_dir_name = (
+        f"{disease.lower()}_r_{report_date}_f_"
+        f"{first_training_date}_t_{last_training_date}"
+    )
+
+    model_data_dir = Path(output_data_dir, model_dir_name)
```

Comment:
Throughout, I have generally used `model_dir` to refer to a particular state folder and `base_dir` to refer to the folder that contains the state folders (each being a `model_dir`). We can change these names (as I don't think they're good), but we should be consistent.

Reply:
I think we should standardize on `model_run_dir` as variable name and `model-run-dir` / `--model-run-dir` as the associated command line argument.